unixfs/mod/dagmodifier.go

```diff
@@ -238,43 +232,87 @@ func (dm *DagModifier) Sync() error {
 // returns the new key of the passed in node and whether or not all the data in the reader
 // has been consumed.
 func (dm *DagModifier) modifyDag(n node.Node, offset uint64, data io.Reader) (*cid.Cid, bool, error) {
-	node, ok := n.(*mdag.ProtoNode)
-	if !ok {
-		return nil, false, ErrNoRawYet
-	}
+	// If we've reached a leaf node.
+	if len(n.Links()) == 0 {
+		switch nd0 := n.(type) {
+		case *mdag.ProtoNode:
+			f, err := ft.FromBytes(nd0.Data())
+			if err != nil {
+				return nil, false, err
+			}
 
-	f, err := ft.FromBytes(node.Data())
-	if err != nil {
-		return nil, false, err
-	}
+			n, err := data.Read(f.Data[offset:])
```

Comment:
might want to use io.ReadFull here.

Reply:
This is not new code so I rather leave it alone.  It is expected that the data from the reader might not completely full the buffer.  If I use ReadFull I will then need to test for both io.EOF and io.ErrUnexpectedEOF, otherwise tests will fail.

Reply:
Ah, i see. fine to not change existing code